addons/web/controllers/main.py

```diff
@@ -651,6 +651,8 @@ def manager(self, **kw):
     @http.route('/web/database/create', type='http', auth="none", methods=['POST'], csrf=False)
     def create(self, master_pwd, name, lang, password, **post):
         try:
+            if not re.match('^[a-zA-Z][a-zA-Z0-9_]+$', name):
+                raise Exception(_('Invalid database name'))
```

Comment:
* be a bit more expansive about what the issue is
* maybe add a [@pattern](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-pattern) to the form field to avoid people sending their crap to the server when possible